src/common/meta/src/cache/container.rs

```diff
@@ -76,12 +81,9 @@ where
     K: Send + Sync,
     V: Send + Sync,
 {
-    async fn invalidate(&self, _ctx: &Context, caches: Vec<CacheIdent>) -> Result<()> {
-        for token in caches
-            .into_iter()
-            .filter(|token| (self.token_filter)(token))
-        {
-            (self.invalidator)(&self.cache, &token).await?;
+    async fn invalidate(&self, _ctx: &Context, caches: &[CacheIdent]) -> Result<()> {
+        for token in caches.iter().filter(|token| (self.token_filter)(token)) {
+            (self.invalidator)(&self.cache, token).await?;
```

Comment:
I think we should call all the invalidators first and then check the `Result`. If we don't, a previous failed invalidator will cause any subsequent ones to not be run.